app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -2559,6 +2562,24 @@ class PluginApi {
   addBulkActionButton(opts) {
     _addBulkButton(opts);
   }
+
+  /**
+   * Allows you add additional search results to the full page search results.
+   * Search results added through this API will be mixed with initial search results
+   * and ranked using the reciprocal ranking fusion algorithm.
+   *
+   * (See `reciprocalRankingAlgorithm()` in `discourse/app/lib/search.js` for more
+   * details on how reciprocal ranking works)
+   *
+   * Full page must be refreshed using `recheckSearchResults()` to see the new results.
```

Comment:
This may be a bit confusing, people may think "Full page" here means a reload is needed. 
```suggestion
   * Use `recheckSearchResults()` to trigger a refresh of the component and see the new results.
```

And on second thought, shouldn't this be automatically done?  I.e. `api.addAdditionalSearchResults` should queue a rerender of the results.

Reply:
> And on second thought, shouldn't this be automatically done? I.e. api.addAdditionalSearchResults should queue a rerender of the results.

Yes this is what I would like to have done automatically. Though, I'm unsure how best to achieve this since the action is inside the controller. Please see https://github.com/discourse/discourse/pull/24202#discussion_r1379230590. I'm wondering how to do this without introducing code smell... Using `getController` and calling the action is bad practice?